pkg/local_object_storage/blobstor/fstree/fstree.go

```diff
@@ -297,6 +299,62 @@ func (t *FSTree) Get(prm common.GetPrm) (common.GetRes, error) {
 	return common.GetRes{Object: obj, RawData: data}, nil
 }
 
+// GetBytes reads object from the FSTree by address into memory buffer in a
+// canonical NeoFS binary format. Returns [apistatus.ObjectNotFound] if object
+// is missing.
+func (t *FSTree) GetBytes(addr oid.Address, opts common.GetBytesOptions) ([]byte, error) {
+	p := t.treePath(addr)
+
+	f, err := os.Open(p)
+	if err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			return nil, logicerr.Wrap(apistatus.ObjectNotFound{})
+		}
+		return nil, fmt.Errorf("open object file %q: %w", p, err)
+	}
+
+	fi, err := f.Stat()
+	if err != nil {
+		return nil, fmt.Errorf("stat object file %q: %w", p, err)
+	}
+	sz := fi.Size()
+	if sz > math.MaxInt {
+		return nil, fmt.Errorf("too big object file %d > %d", sz, math.MaxInt)
+	}
+	if sz == 0 {
+		return nil, nil
+	}
+
+	var b []byte
+	if opts.GetBuffer != nil {
+		b = opts.GetBuffer(int(sz))
+	} else {
+		b = make([]byte, sz)
+	}
+
+	_, err = io.ReadFull(f, b)
```

Comment:
why not streaming?

Reply:
u mean return `io.Reader`? no need for now, may be added later

Reply:
we have big problems with memory, i thought that extending the storage interface with row data should be done for memory optimizations mainly

Reply:
> for memory optimizations

it is: as i mentioned in the commit, in some cases we wanna skip decode-encode round

Reply:
i mean reading a full object in memory at once is not good to me in a PR that tries to beat memory problems